docs/del3/Ytelse.md

```diff
@@ -1 +1,166 @@
+https://github.com/tsenart/vegeta/releases. Husk å velge riktig fil til din
+plattform.
+
+### Bruk
+
+Vegeta har to moduser: attack og report. Angrip (`attack`) gjør kall til
+url(ene) man oppgir og `report` generer rapporter basert på informasjonen fra
+attack-kommandoen.
+
+Se Vegetas readme for detaljer utover det som forklares i denne guiden:
+https://github.com/tsenart/vegeta#-reporter
+
+#### Attack-kommandoen
+
+Attack-kommandoen utfører selve ytelsestesten. Her angir man parametrene for
+testen ved å angi url(ene) å gå mot, antall spørringer i sekundet og hvor lenge
+testen skal kjøre.
+
+Outputen av attack kommadoen er binærtdata som man kan lagre til fil eller
+gi direkte videre til Vegetas report-kommando.
+
+Den følgende kommandoen vil sende 5 `GET`-spørringer i sekundet i 10 minutter til
+urlen som er spesifisert og få ut en tekstbasert rapport på hvor lang tid
+kallene tok og annen info om kjøringen av ytelsestesten:
+
+```
+echo "GET https://5xv110usmk.execute-api.eu-central-1.amazonaws.com/dev/todos" | vegeta attack -rate 5 -duration 10m -http2=f | vegeta report
+```
+
+
+#### Report-kommandoen
+
+Report-kommandoen til Vegeta genererer rapporter basert på output fra attack-kommandoen.
+Basert på denne binærdataen kan rapport-kommandoen generere
+forkjellige rapporter. Rapportene som kommer innebyggd er: text, json, histogram
+og plot. Vi kommer nok kun til å trenge tekst-versjonen som er standard om man
+ikke oppgir noe annet.
+
+Se Vegetas readme for mer informasjon om reporterne: https://github.com/tsenart/vegeta#-reporter.
+
+
+
+## Første ytelsestest og utforsking
+
+- Sett i gang en ytelsestest med Vegeta på 40 spørringer i sekundet over 10
+  minutter. Det kan ta noen minutter før man får utslag i grafene, så bare start
+  kommandoen og gå videre til punktenen under.
+- Gå deretter til DynamoDB-tabellen for applikasjonen din i AWS-konsollet og gå
+  til fanen _Metrics_. Her kan du se aktiviteten for tabellen i forskjellige
+  grafer. Blant annet ser man antall lesninger og skrivinger i sekundet og
+  ventetid (latency) for lesing og skriving mm. Det er desverre ikke noen
+  automatisk oppdatering av grafene så du må selv trykke på oppdateringskappen
+  oppe til høyret i Metrics vinduet.
+- Sett _Time Range_ oppe til høyre på siden til _Last Hour_ for å se de nyeste
+  hendelsene. Når ytelsestesten har kjørt noen minutter vil man kunne begynne å
+  se utslag på grafene. Prøv med forskjellige antall spørringer i sekundet og se
+  hvordan det påvirker ytelsen til databasen.
+- Alt etter hvordan du konfigurerte DynamoDB-tabellen i de tidligere oppgavene
+  vil du nå et punkt hvor spørringer blir blokkert. Disse vil du se i grafene
+  «Throttled read requests» og «Throttled write requests».
+
+>Hvis man har satt «Provisioned read capacity units» til 1 vil man kunne gjøre
+én lesing i sekundet (pluss en viss mengde over i kortere perioder). Hvis man
+utfører en test med 5 spørringer i sekundet vil man da være over grensen og om
```

Comment:
I starten så setter vi i gang en test med 40 spørringer i sekundet, mens det her snakkes om 5 spørringer i sekundet? Bør vi konsekvent si 5 eller 40?

Reply:
Kan være lurt å være konsekvent. Poenget med å bruke 40 er at det gir et mye raskere og tydeligere utslag i loggene. Endrer fra 5 til 40 for å være konsekvent.